imap_processing/ena_maps/utils/spatial_utils.py

```diff
@@ -0,0 +1,270 @@
+    # We can infer the shape if its evenly spaced and 2D
+    if not shape:
+        spacing_deg = 1 / np.sqrt(raveled_values.shape[0] / (360 * 180))
+        shape = (int(180 // spacing_deg), int(360 // spacing_deg))
+
+    if extra_axis:
+        shape = (shape[0], shape[1], raveled_values.shape[1])
+    return raveled_values.reshape(shape, order=order)
+
+
+class AzElSkyGrid:
+    """
+    Representation of a 2D grid of azimuth and elevation angles covering the sky.
+
+    All angles are stored internally in radians, but can be accessed in degrees, by
+    appending "_degrees" to the attribute name. For example, `spacing` in radians
+    can be accessed as `spacing_degrees` in degrees.
+
+    Parameters
+    ----------
+    spacing_deg : float, optional
+        Spacing of the grid in degrees, by default 0.5.
+    centered_azimuth : bool, optional
+        Whether the azimuth grid should be centered around 0 degrees/0 radians,
+        i.e. from -pi to pi radians, by default False.
+        If True, the azimuth grid will be from -pi to pi radians.
+    centered_elevation : bool, optional
+        Whether the elevation grid should be centered around 0 degrees/0 radians,
+        i.e. from -pi/2 to pi/2 radians, by default True.
+        If False, the elevation grid will be from 0 to pi radians.
+    reversed_elevation : bool, optional
+        Whether the elevation grid should be reversed, by default False.
+        If False, the elevation grid will be from -pi/2 to pi/2 radians (-90 to 90 deg).
+        If True, the elevation grid will be from pi/2 to -pi/2 radians (90 to -90 deg).
+
+    Raises
+    ------
+    ValueError
+        If the spacing is not positive or does not divide evenly into pi radians.
+    """
+
+    def __init__(
+        self,
+        spacing_deg: float = 0.5,
+        centered_azimuth: bool = False,
+        centered_elevation: bool = True,
+        reversed_elevation: bool = False,
+    ) -> None:
+        # Store grid properties
+        self.centered_azimuth = centered_azimuth
+        self.centered_elevation = centered_elevation
+        self.reversed_elevation = reversed_elevation
+
+        # Internally, work in radians, regardless of desired output units
+        # If angular_units == deg, conversion will be done at the end
+        self.spacing = np.deg2rad(spacing_deg)
+
+        # Ensure valid grid spacing (positive, divides evenly into pi radians)
+        if self.spacing <= 0:
+            raise ValueError("Spacing must be positive valued, non-zero.")
+
+        if not np.isclose((np.pi / self.spacing) % 1, 0):
+            raise ValueError("Spacing must divide evenly into pi radians.")
+
+        # build_spacial_bins creates the bin edges and centers for azimuth and elevation
+        # E.g. for spacing=1, az_bin_edges = [0, 1, 2, ..., 359, 360] deg.
+        (az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints) = (
+            build_spatial_bins(az_spacing=spacing_deg, el_spacing=spacing_deg)
```

Comment:
Below this you call `np.deg2rad` on all these return values. Maybe the helper function should just return radians instead?

Reply:
Yeah, that makes sense. I was trying to keep it consistent with how `build_spatial_bins` was made for Laura's L1 code, which uses degrees.

I can switch this to returning radians, and just adjust the L1a code to convert